File: examples/ruby/advanced/03_functional.rb
# Ruby Advanced: Enumerable (Functional)

numbers = [1, 2, 3, 4, 5]

# Map
squares = numbers.map { |n| n * n }
puts "Squares: #{squares}"

# Select (Filter)
evens = numbers.select { |n| n.even? }
puts "Evens: #{evens}"

# Reduce
sum = numbers.reduce(0) { |acc, n| acc + n }
puts "Sum: #{sum}"

File: examples/python/advanced/03_functional.py
# Python Advanced: Functional Tools

numbers = [1, 2, 3, 4, 5]

# List comprehension
squares = [n * n for n in numbers]
print("Squares:", squares)

# Filter
evens = list(filter(lambda x: x % 2 == 0, numbers))
print("Evens:", evens)

# Map
double = list(map(lambda x: x * 2, numbers))
print("Double:", double)
